cmd/admin/pause_identifier.go

```diff
@@ -39,39 +43,78 @@ func (p *subcommandPauseIdentifier) Run(ctx context.Context, a *admin) error {
 		return err
 	}
 
-	_, err = a.pauseIdentifiers(ctx, identifiers)
+	_, err = a.pauseIdentifiers(ctx, identifiers, p.maxInFlight)
 	if err != nil {
 		return err
 	}
 
 	return nil
 }
 
-// pauseIdentifiers allows administratively pausing a set of domain names for an
-// account. It returns a slice of PauseIdentifiersResponse or an error.
-func (a *admin) pauseIdentifiers(ctx context.Context, incoming []pauseCSVData) ([]*sapb.PauseIdentifiersResponse, error) {
-	if len(incoming) <= 0 {
+// pauseIdentifiers pauses each account, identifier pair in the provided slice
+// of pauseCSVData entries. It will pause up to maxInFlight identifiers at a
+// time. If any errors occur while pausing, they will be gathered and returned
+// as a single error.
+func (a *admin) pauseIdentifiers(ctx context.Context, entries []pauseCSVData, maxInFlight int64) ([]*sapb.PauseIdentifiersResponse, error) {
+	if len(entries) <= 0 {
 		return nil, errors.New("cannot pause identifiers because no pauseData was sent")
 	}
 
-	var responses []*sapb.PauseIdentifiersResponse
-	for _, data := range incoming {
-		req := sapb.PauseRequest{
-			RegistrationID: data.accountID,
-			Identifiers: []*sapb.Identifier{
-				{
-					Type:  string(data.identifierType),
-					Value: data.identifierValue,
-				},
-			},
-		}
-		response, err := a.sac.PauseIdentifiers(ctx, &req)
-		if err != nil {
-			return nil, err
-		}
+	accountToIdentifiers := make(map[int64][]*sapb.Identifier)
+	for _, entry := range entries {
+		accountToIdentifiers[entry.accountID] = append(accountToIdentifiers[entry.accountID], &sapb.Identifier{
+			Type:  string(entry.identifierType),
+			Value: entry.identifierValue,
+		})
+	}
+
+	respChan := make(chan *sapb.PauseIdentifiersResponse, len(accountToIdentifiers))
+	errorsChan := make(chan error, len(accountToIdentifiers))
+	sem := semaphore.NewWeighted(maxInFlight, 0)
```

Comment:
Note that we already have a parallelism implementation for [key blocking](https://github.com/letsencrypt/boulder/blob/da7865cb107bf96a5bd4c3643f38066dafba2233/cmd/admin/key.go#L158) and [cert revoking](https://github.com/letsencrypt/boulder/blob/da7865cb107bf96a5bd4c3643f38066dafba2233/cmd/admin/cert.go#L281), and I think that the method used by both of those is much more efficient. It only uses a single channel whose buffer size doesn't depend on the input size, and it only spins up M goroutines instead of spinning up N goroutines and then letting them compete for M semaphores.

Now obviously efficiency isn't the highest priority for something like the admin tool. But I think that, unless we have a strong reason to believe that this implementation has specific benefits, it's better to stick with the same structure as already used by other admin subcommands.